geoutils/georaster.py

```diff
@@ -941,15 +953,15 @@ def reproject(
                 (dst_res == self.res) or (dst_res == self.res[0] == self.res[1]) or (dst_res is None),
             ]
         ):
-            if nodata == self.nodata:
+            if (dst_nodata == self.nodata) or (dst_nodata is None):
                 if not silent:
                     warnings.warn("Output projection, bounds and size are identical -> return self (not a copy!)")
                 return self
 
-            else:
+            elif dst_nodata is not None:
                 warnings.warn("Only nodata is different, running self.set_ndv instead")
```

Comment:
I know it was not introduced in this particular PR, but this kind of reads like it will return self, but will in fact return a copy. Could this be reformulated appropriately?

Gets me thinking, there should be a method called `Raster.mirror(other_raster)` or something similar that will reprpject (if needed), nodata, resolution, bounds, dtype(?) etc. without warning about if it's actually needed or not! Should I add this as an issue maybe? Basically a function that does: "make sure that these two rasters can now be directly compared".

Reply:
Good point... What do you think, should it rather return self? Or leave it as it is but rephrase the warning?

Reply:
Regarding the mirror method, I feel like this overlaps almost exactly with reproject, except for the nodata value. But is there any reason to absolutely want the two rasters to have the same nodata value?
In any case, you would get what you want with just:
`raster.reproject(dst_ref=raster_ref, dst_nodata=raster_ref.nodata)`
which seems simple enough for not having a dedicated function for it.

Reply:
I would do something like:

```python
"Only nodata is different. Consider using the 'set_ndv()' method instead."
```
if this warning is even necessary (I don't personally).

Also, I think this warning should respect the `silent` flag.

Reply:
Regarding the mirror, nevermind. Because as you say, there's no use for synchronizing all properties that I know of!